src/middleware.ts

```diff
@@ -0,0 +1,15 @@
+import { NextResponse } from 'next/server'
+
+export function middleware(request: Request) {
+  const url = new URL(request.url)
+  const pathname = url.pathname
+
+  const requestHeaders = new Headers(request.headers)
+  requestHeaders.set('x-pathname', pathname)
```

Comment:
It's necessary to pass the pathname from middleware this way so that the `markdoc/node` file can match up the navigation item.